storm-core/src/clj/backtype/storm/util.clj

```diff
@@ -1062,6 +1062,22 @@
 (defn hashmap-to-persistent [^HashMap m]
   (zipmap (.keySet m) (.values m)))
 
+(defn retry-on-exception
+  "Retries specific function on exception based on retries count"
+  [retries task-description f & args]
+  (let [res (try {:value (apply f args)}
+              (catch Exception e
+                (if (= 0 retries)
```

Comment:
Would like to check for `(not (pos?` or `<= 0` here in case we get a negative value for `retries`. We would get a LOT of log messages until in underflowed and came back to 0.